### PR TITLE
gnulib: 20210702 -> 20231109

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

https://github.com/coreutils/gnulib/compare/901694b904cd861adc2529b2e05a3fb33f9b534f...2dd1a7984c6b3e6056cef7e3f9933e0039c21634 (:warning: huge changelog)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 7c6f434c** [comment]
@ofborg build wget2

**Comment by anthonyroussel** [comment]
@ofborg build gnulib wget2 xprintidle-ng

**Comment by anthonyroussel** [comment]
Fixed failing wget2 build on Darwin

File: pkgs/development/tools/gnulib/default.nix [hunk 1]
```diff
@@ -1,13 +1,13 @@
 { lib, stdenv, fetchgit, python3 }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "gnulib";
-  version = "20210702";
+  version = "20231109";
 
   src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gnulib/default.nix:7]: Can we use fetchFromSavannah here?

File: pkgs/development/tools/gnulib/default.nix [hunk 2]
```diff
@@ -40,6 +40,7 @@ stdenv.mkDerivation {
   meta = with lib; {
     description = "Central location for code to be shared among GNU packages";
     homepage = "https://www.gnu.org/software/gnulib/";
+    changelog = "https://git.savannah.gnu.org/gitweb/?p=${pname}.git;a=blob;f=ChangeLog";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gnulib/default.nix:43]:
```suggestion
    changelog = "https://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=blob;f=ChangeLog";
```

File: pkgs/development/tools/gnulib/default.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/development/tools/gnulib/default.nix]: Fixed, thanks for the review

**Comment by 7c6f434c** [comment]
@ofborg eval

**Comment by anthonyroussel** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 266557` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>gnulib</li>
<li>idutils</li>
<li>lbzip2</li>
<li>lua51Packages.lrexlib-gnu</li>
<li>luaPackages.lrexlib-gnu (lua52Packages.lrexlib-gnu)</li>
<li>lua53Packages.lrexlib-gnu</li>
<li>lua54Packages.lrexlib-gnu</li>
<li>luajitPackages.lrexlib-gnu</li>
</ul>
</details>

**Pull request merged**